### ISSUE TITLE
Add method to BitArray for converting to bit array

### ISSUE THREAD

**Issue by kevinsung** [Qiskit/qiskit#13396]
### What should we add?

I.e., a Numpy array of bools. Recipe courtesy of @ihincks:
```python
np.unpackbits(bit_array.array, axis=-1)[...,-bit_array.num_bits:]
```

**Comment by boonware** [comment]
@kevinsung Could you elaborate on this one a little more, I am interested in implementing it.

**Comment by mohitd** [comment]
`BitArray` has [`from_bool_array`](https://github.com/Qiskit/qiskit/blob/4bb143275cfb01954d2898c44afe48e08640bcad/qiskit/primitives/containers/bit_array.py#L193) function so is the intent of this feature request to create a corresponding `to_bool_array`?

**Comment by ihincks** [comment]
@mohitd yes, exactly. It is not a difficult implementation, see the original post, but it is inconvenient to remember about unpack bits and the convention of which side of the array needs to have the padding discarded, etc.

**Comment by amiigas** [comment]
Hi @jakelishman, I can add this if nobody is taking a look, could you assign me please?

**Comment by mohitd** [comment]
@ihincks Makes sense. 

@amiigas All yours :) I have some available time coming up so let me know if you'll pick this up

**Comment by amiigas** [comment]
@mohitd Thanks, I will!

**Comment by kevinsung** [comment]
Closed in https://github.com/Qiskit/qiskit/pull/13653
